ampel/contrib/hu/t3/PlotTransientLightcurves.py

```diff
@@ -0,0 +1,615 @@
+
+        info.append(name)
+        info.append(f"RA: {ra:.8f}")
+        info.append(f"Dec: {dec:.8f}")
+        info.append("------------------------")
+
+        fig.text(0.77, 0.55, "\n".join(info), va="top", fontsize="medium", alpha=0.5)
+
+    # Add annotations
+    # Frits
+    if fritzlink:
+        lc_ax1.annotate(
+            "See On Fritz",
+            xy=(0.5, 1),
+            xytext=(0.78, 0.10),
+            xycoords="figure fraction",
+            verticalalignment="top",
+            color="royalblue",
+            url=f"https://fritz.science/source/{name}",
+            fontsize=12,
+            bbox=dict(boxstyle="round", fc="cornflowerblue", ec="royalblue", alpha=0.4),
+        )
+    # TNS
+    if tnsname is not None:
+        lc_ax1.annotate(
+            "See On TNS",
+            xy=(0.5, 1),
+            xytext=(0.78, 0.05),
+            xycoords="figure fraction",
+            verticalalignment="top",
+            color="royalblue",
+            url=f"https://www.wis-tns.org/object/{tnsname}",
+            fontsize=12,
+            bbox=dict(boxstyle="round", fc="cornflowerblue", ec="royalblue", alpha=0.4),
+        )
+
+    # Catalog info through T2CatalogMatch results. Add to info above, or keep some of it? At least check TNS somehow
+    if len(attributes) > 0:
+        ypos = 0.975 if cutouts else 0.035
+        fig.text(
+            0.5,
+            ypos,
+            " - ".join(attributes),
+            va="top",
+            ha="center",
+            fontsize="medium",
+            alpha=0.5,
+        )
+
+    if t_0_jd is not None:
+        lc_ax1.axvline(t_0_jd, linestyle=":")
+    else:
+        t_0_jd = np.mean(fluxtable["time"])
+
+    # Ugly hack because secondary_axis does not work with astropy.time.Time datetime conversion
+    jd_min = min(np.min(fluxtable["time"]), t_0_jd)
+    if ztfulims is not None:
+        jd_min = min(np.min(ztfulims["time"]), jd_min)
+    jd_max = max(np.max(fluxtable["time"]), t_0_jd)
+    length = jd_max - jd_min
+
+    lc_ax1.set_xlim((jd_min - (length / 20), jd_max + (length / 20)))
+
+    lc_ax2 = lc_ax1.twiny()
+
+    lc_ax2.scatter(  # type: ignore
+        [Time(x, format="jd").datetime for x in [jd_min, jd_max]], [20, 20], alpha=0
+    )
+    lc_ax2.tick_params(axis="both", which="major", labelsize=6, rotation=45)
+    lc_ax1.tick_params(axis="x", which="major", labelsize=6, rotation=45)
+    lc_ax1.ticklabel_format(axis="x", style="plain")
+    lc_ax1.tick_params(axis="y", which="major", labelsize=9)
+
+    if z is not None:
+        lc_ax3.tick_params(axis="both", which="major", labelsize=9)
+
+    #    if z is not None:
+    #        axes = [lc_ax1, lc_ax2, lc_ax3]
+    #    else:
+    #        axes = [lc_ax1, lc_ax2]
+    axes = [lc_ax1, lc_ax2, lc_ax3] if z is not None else [lc_ax1, lc_ax2]
+
+    return fig, axes
+
+
+@backoff.on_exception(
+    backoff.expo,
+    requests.exceptions.RequestException,
+    max_time=600,
+)
+def ampel_api_cutout(
```

Comment:
This also looks like it duplicates the functionality of [ZTFCutoutImages](https://github.com/AmpelAstro/Ampel-ZTF/blob/662e8174685d3793c85a14d278362982f14ed8bd/ampel/ztf/t3/complement/ZTFCutoutImages.py)

Reply:
Yes, so now the main mode of PlotTransientLightcurves uses ZTFCutoutImages. However I now left the direct retrieval of thumbnails in the unit as a potential work mode. That option can be easily removed if we want to easy maintenance. But I left it there to possibly make it usable in isolation? Also, how to use complement is not straightforward to figure out...